pkg/queue/manager.go

```diff
@@ -105,13 +105,28 @@ func (m *Manager) DeleteQueue(q *kueue.Queue) {
 	delete(m.queues, key)
 }
 
-func (m *Manager) Status(q *kueue.Queue) (int, error) {
+func (m *Manager) Status(q *kueue.Queue) (map[string]int32, error) {
 	m.RLock()
 	defer m.RUnlock()
-	if q := m.queues[Key(q)]; q != nil {
-		return q.heap.Len(), nil
+
+	jobs := make(map[string]int32)
+	queue, ok := m.queues[Key(q)]
+	if !ok {
+		return jobs, errQueueDoesntExist
 	}
-	return 0, errQueueDoesntExist
+	jobs["pending"] = int32(queue.heap.Len())
+
+	var admitted int32
+	for _, hi := range queue.heap.items {
```

Comment:
An admitted workload wouldn't exist in the queue at all. We will need to list the workloads using the index and count the assigned ones (unless we somehow keep track of them in the Queue data structure).

Reply:
and listing is going to have perf implications, I don't think we should do it. Tracking number of admitted workloads at the clusterQueue level will probably be good enough for now.

How about we use this PR to only update the test as discussed in the other PR?

Reply:
Good point, I am a bit worried on the nested loop as well, but we do need something to query the queue it self, and not at the cluster lvl. 
users might want to submit several workloads, and watch for their progress

Reply:
It is useful in both places, admins would be interested in checking the number of workloads admitted to a ClusterQueue. 

> users might want to submit several workloads, and watch for their progress

right, but only if we can do that in an efficient way

Reply:
oh for admins I agree 100%.

@alculquicondor maybe you have an idea on how to query this without nesting?

Reply:
I am not sure what you mean by nesting, but as I mentioned above, admitted workloads don't exist in `queue.heap.items` at all; the queue only have the workloads to be scheduled (pending)

Reply:
what we need to do is perhaps organize the admitted workloads in each CQ broken by Q name, that is the most efficient way I believe.

Reply:
If Q's are going to be namespaced, users need a way to fully manage their QW and Q's at their permission lvl 
never ever have to interact with CQ's

Reply:
>How about we use this PR to only update the test as discussed in the other PR?

Agree, done

Reply:
>How about we use this PR to only update the test as discussed in the other PR?

Agree, done